src/core/lib/security/credentials/credentials.cc

```diff
@@ -45,6 +45,14 @@ void grpc_channel_credentials_release(grpc_channel_credentials* creds) {
   if (creds) creds->Unref();
 }
 
+grpc_channel_credentials* grpc_channel_credentials_copy(
```

Comment:
@HannahShiSFB Can you please comment on why we need to introduce this new function in c-core? Please also comment / give example to why we find it consistent to call the function `_copy` with other similar existing functions. Thanks.

Reply:
The goal is adding a copy(ref) of the grpc creds object.
Since the grpc creds object is created by php within channel object, when php destroy its channel object, if we do nothing,  the grpc creds will be destroyed with it. 
To avoid the grpc creds be destroyed then, since we still need it later(when fork), we introduce a c-core API to add a copy(ref) of the grpc creds object. 
Adding a c-core API is the only way to call creds->Ref().release() in it to add a copy(ref).

Q: why we name it _copy instead of _ref? 
A: several other similar existing functions with Ref().release() named this way. eg, credentials_pointer_arg_copy(), target_authority_table_copy(), channelz_node_copy(), grpc_channel_args_copy(), etc.

Q: why we can only call Ref().release() to add a copy(ref)?
A: adding a copy(ref) equals count++. cpp can automatically add 1 when we put creds obj in persistent list, but in c, it's just a pointer, we have to add 1 by ourselves. Since we cannot access the count directly, cannot simply call count++, so we use Ref().release() to get there.
How? in ref_counted.h:L253 Ref() returns RefCountedPtr. release() is to make sure when RefCountedPtr distruct, count-- will not happen. When construct count will +1, when distruct, count remains its value, eventually we made count++. 

Reply:
Overall I am fine with adding a copy function in  `grpc_security.h`, if that's the only way to solve the PHP problem mentioned in this PR. I have some follow-up questions on this thread:

1. May I know why we couldn't make use of  the existing `credentials_pointer_arg_copy`,  and expose it in `grpc_security.h`? From the documentation above, It seems this function is doing the same as the newly added `grpc_channel_credentials_copy`. Does `grpc_core::ExecCtx` make any difference here, or you want to add some extra logging message?
2.  Is this issue only applicable to the channel credential? shall we also add it for the server credential?

Reply:
@ZhenLian Thanks for the review.
1. I will look into `credentials_pointer_arg_copy` and get back to you
2. PHP doesn't need server credentials in this case. I will leave it up to your judgement after this is settled.

Thanks!

Reply:
@ZhenLian We are open to c-core exposing `credentials_pointer_arg_copy` as an API - do you have concern that the current implementation didn't check for the validity of `void* p`? If you don't have concern, we will be happy to change this PR to use `channel_pointer_arg_copy` instead. 

Do you want us to submit a separate PR just to expose `credentials_pointer_arg_copy` in `grpc_security.h`?

Thanks again.

Reply:
Sorry for the slight delay! I somehow missed my email notifications. 
Yeah I think in general it would be great to add C core changes in a separate PR, and adding the validity check there. Thanks for making the changes!

Reply:
@ZhenLian Thanks. We will work on another PR just for that `credentials_pointer_arg_copy` piece and ask for your review again. Thanks!